packages/nitro-protocol/contracts/ninja-nitro/SingleChannelAdjudicator.sol

```diff
@@ -158,6 +158,16 @@ contract SingleChannelAdjudicator is
         emit Concluded(channelId, uint48(block.timestamp)); //solhint-disable-line not-rely-on-time
     }
 
+    /**
+     * @notice Checks if a given destination is external (i.e. simply a padded Ethereum addtess), or a channelId (which has a corresponding create2 address)
+     * @dev Checks if a given destination is external (i.e. simply a padded Ethereum addtess), or a channelId (which has a corresponding create2 address)
+     * @param destination Destination to be checked.
+     * @return True if the destination is external, false otherwise.
+     */
+    function _isExternalDestination(bytes32 destination) internal pure returns (bool) {
+        return uint96(bytes12(destination)) == 0;
+    }
```

Comment:
Isn't this a core Nitro question? I feel like there should be a single implementation of this in `nitro-protocol`, and that it shouldn't live in `ninja-nitro`.

Reply:
The implementation lives in the `AssetHolder` contract currently. Everything that is currently in that contract and is needed in some form in`ninja-nitro`, has been copypasted.

 Perhaps it should live in the  `Outcome` library eventually, where it can be imported by both implementations and not replicated.